src/SixLabors.Fonts/FontRectangle.cs

```diff
@@ -0,0 +1,374 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace SixLabors.Fonts
+{
+    /// <summary>
+    /// Stores a set of four single precision floating points that represent the location and size of a rectangle.
+    /// </summary>
+    /// <remarks>
+    /// This struct is fully mutable. This is done (against the guidelines) for the sake of performance,
+    /// as it avoids the need to create new values for modification operations.
+    /// </remarks>
+    public readonly struct FontRectangle : IEquatable<FontRectangle>
+    {
+        /// <summary>
+        /// Represents a <see cref="FontRectangle"/> that has X, Y, Width, and Height values set to zero.
+        /// </summary>
+        public static readonly FontRectangle Empty = default;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FontRectangle"/> struct.
+        /// </summary>
+        /// <param name="x">The horizontal position of the rectangle.</param>
+        /// <param name="y">The vertical position of the rectangle.</param>
+        /// <param name="width">The width of the rectangle.</param>
+        /// <param name="height">The height of the rectangle.</param>
+        public FontRectangle(float x, float y, float width, float height)
+        {
+            this.X = x;
+            this.Y = y;
+            this.Width = width;
+            this.Height = height;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FontRectangle"/> struct.
+        /// </summary>
+        /// <param name="point">
+        /// The <see cref="Point"/> which specifies the rectangles point in a two-dimensional plane.
+        /// </param>
+        /// <param name="size">
+        /// The <see cref="Size"/> which specifies the rectangles height and width.
+        /// </param>
+        internal FontRectangle(Vector2 point, Vector2 size)
```

Comment:
It looks like we don't need this as an internal constructor? Maybe make it private then instead?

Reply:
your right, made it private.